hydra_base/lib/template.py

```diff
@@ -1526,22 +1585,35 @@ def _set_typeattr(typeattr, existing_ta = None):
 
     ta.data_restriction = _parse_data_restriction(typeattr.data_restriction)
 
-    if typeattr.dimension is not None and typeattr.attr_id is not None and typeattr.attr_id > 0:
-        attr = ta.attr
-        if attr.dimension != typeattr.dimension:
-            raise HydraError("Cannot set a dimension on type attribute which "
-                            "does not match its attribute. Create a new attribute if "
-                            "you want to use attribute %s with dimension %s"%
-                            (attr.name, typeattr.dimension))
-    elif typeattr.dimension is not None and typeattr.attr_id is None and typeattr.name is not None:
-        attr = _get_attr_by_name_and_dimension(typeattr.name, typeattr.dimension)
-        ta.attr_id = attr.id
-        ta.attr = attr
+    if typeattr.dimension_id is None:
+        # All right. Check passed
+        pass
+    else:
+
+        if typeattr.attr_id is not None and typeattr.attr_id > 0:
+            # Getting the passed attribute, so we need to check consistency between attr dimension id and typeattr dimension id
+            attr = ta.attr
+
+            if attr is not None and attr.dimension_id is not None and attr.dimension_id != typeattr.dimension_id or \
+               attr is not None and attr.dimension_id is not None:
+                # In this case there is an inconsistency between attr.dimension_id and typeattr.dimension_id
```

Comment:
Good comments. These comments are really useful.

Reply:
Good to know. :)
Thanks